src/sqlancer/StatementReducer.java

```diff
@@ -28,32 +33,59 @@ public void reduce(G state, Reproducer<G> reproducer, G newGlobalState) throws E
         printQueries(knownToReproduceBugStatements);
         System.out.println();
 
-        do {
+        if (knownToReproduceBugStatements.size() <= 1) {
+            return;
+        }
+
+        partitionNum = 2;
+        while (knownToReproduceBugStatements.size() >= 2) {
             observedChange = false;
+
             knownToReproduceBugStatements = tryReduction(state, reproducer, newGlobalState,
-                    knownToReproduceBugStatements, (candidateStatements, i) -> {
-                        candidateStatements.remove((int) i);
+                    knownToReproduceBugStatements, (candidateStatements, s, l) -> {
```

Comment:
I'd suggest that we use the variables `startIndex` and `subLength` for better code readability. But that's just my opinion, we can discuss with @mrigger